torch/csrc/jit/frontend/source_ref.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include <c10/macros/Export.h>
+#include <torch/csrc/jit/frontend/source_range.h>
+
+namespace torch {
+namespace jit {
+
+class TORCH_API SourceRef {
+ public:
+  explicit SourceRef(std::shared_ptr<Source> source)
+      : source_(std::move(source)) {}
+  bool operator==(const SourceRef& other) const {
+    return source_ == other.source_;
+  }
+  bool operator<(const Source& other) const {
+    return source_.get() < &other;
+  }
+  friend bool operator<(const Source& other, const SourceRef& self) {
+    return &other < self.source_.get();
+  }
+  bool operator<(const SourceRef& other) const {
+    return *this < *other.source_.get();
+  }
```

Comment:
Curious why these comparators are needed.

Reply:
I need to put this type as keys in std::map, and std::map is comparison based.